VSDiagnostics/VSDiagnostics/VSDiagnostics/Utilities/Extensions.cs

```diff
@@ -144,5 +145,37 @@ public static bool IsAsync(this IMethodSymbol methodSymbol)
                 || methodSymbol.ReturnType.MetadataName == typeof(Task).Name
                 || methodSymbol.ReturnType.MetadataName == typeof(Task<>).Name;
         }
+
+        public static bool IsDefinedInAncestor(this IMethodSymbol methodSymbol)
+        {
+            var containingType = methodSymbol?.ContainingType;
+            if (containingType == null)
+            {
+                return false;
+            }
+
+            var interfaces = containingType.AllInterfaces;
+            foreach (var @interface in interfaces)
+            {
+                var interfaceMethods = @interface.GetMembers().Select(containingType.FindImplementationForInterfaceMember);
+                if (interfaceMethods.Any(method => method.Equals(methodSymbol)))
+                {
+                    return true;
+                }
+            }
+
+            var baseType = containingType.BaseType;
+            while (baseType != null)
```

Comment:
Merged! 

Don't worry about it, we all are ;-) I've had to put it on the backburner for a time as well since I'm interning now but as soon as I've graduated in two months, I can crank it up a notch again. Whatever you can contribute is already highly appreciated